Packs/AWS_SystemManager/Integrations/AWSSystemManager/AWSSystemManager.py

```diff
@@ -0,0 +1,1308 @@
+            )
+        else:
+            return PollResult(  # CONTINUE POLLING
+                partial_result=CommandResults(
+                    readable_output=f"Execution {command_id} is {status}",
+                ),
+                continue_to_poll=True,
+                args_for_next_run=args,
+                response=None,
+            )
+
+    # Initial command execution
+    kwargs = {"CommandId": command_id}
+    if instance_ids := argToList(args.get("instance_ids")):
+        kwargs["InstanceIds"] = argToList(instance_ids)
+    ssm_client.cancel_command(**kwargs)
+    args["first_run"] = False
+    status = get_command_status(command_id, ssm_client)
+
+    return PollResult(
+        response=CommandResults(
+            readable_output="Cancellation command was sent successful.",
+        ),  # if the polling is stop after first run, this will be the final result in the war room
+        partial_result=CommandResults(
+            readable_output="Cancellation command was sent successful.",
+        ),  # if the polling is not stop after first run, this will be the partial result
+        continue_to_poll=include_polling,
+    )
+
+
+def test_module(ssm_client: "SSMClient") -> str:
+    """Tests the connectivity to AWS Systems Manager (SSM) by listing associations.
+
+    Args:
+    ----
+        ssm_client (SSMClient): An instance of the AWS Systems Manager (SSM) client.
+
+    Returns:
+    -------
+        str: A status message indicating the success of the test.
+    """
+    ssm_client.list_associations(MaxResults=1)
+    return "ok"
+
+
+def main():
+    params = demisto.params()
+    command = demisto.command()
+    args = demisto.args()
+    verify_certificate = not params.get("insecure", True)
+
+    aws_access_key_id = params["credentials"]["identifier"]
+    aws_secret_access_key = params["credentials"]["password"]
+    aws_default_region = params["defaultRegion"]
+    aws_role_arn = params.get("roleArn")
+    aws_role_session_name = params.get("roleSessionName")
+    aws_role_session_duration = params.get("sessionDuration")
+    aws_role_policy = None  # added it for using AWSClient class without changing the code
+    timeout = params["timeout"]
+    retries = params["retries"]
+
+    validate_params(
+        aws_default_region,
+        aws_role_arn,
+        aws_role_session_name,
+        aws_access_key_id,
+        aws_secret_access_key,
+    )
+
+    aws_client = AWSClient(
+        aws_default_region,
+        aws_role_arn,
+        aws_role_session_name,
+        aws_role_session_duration,
+        aws_role_policy,
+        aws_access_key_id,
+        aws_secret_access_key,
+        verify_certificate,
+        timeout,
+        retries,
+    )
+
+    ssm_client = config_aws_session(args, aws_client)
+
+    demisto.debug(f"Command being called is {command}")
+    try:
+        match command:
+            case "test-module":
+                return_results(test_module(ssm_client))
```

Comment:
do not call `return_results` in each line, assign `results = ...` and then, in the end, `return_results(results)`

Reply:
Done